shark-hprof/src/main/java/shark/Hprof.kt

```diff
@@ -40,7 +39,7 @@ class Hprof private constructor(
     if (currentPosition == newPosition) {
       return
     }
-    source.buffer.clear()
+    source.buffer().clear()
```

Comment:
You should use the latest 1.x release. In that you'll see this is deprecated; replaced with getBuffer().

Reply:
Version 1.17.4.

Reply:
Oh and in great irony the bug report is from a user on a version that precedes this one. That is, this bug can occur exclusively within 1.x releases. My recommendation is to decline this PR and tell your user to use the latest 1.x.

Reply:
buffer vs getBuffer will work for 1.17.4, but I expect the extension functions weren't available then.